### ISSUE TITLE
DNS server address without scheme won't work

### ISSUE THREAD

**Issue by zhuzeitou** [SagerNet/sing-box#1498]
### Operating system

Windows

### System version

Windows 11 23H2

### Installation type

Original sing-box Command Line

### Version

```shell
sing-box version 1.9.0-alpha.14

Environment: go1.22.0 windows/amd64
Tags: with_gvisor,with_quic,with_dhcp,with_wireguard,with_ech,with_utls,with_reality_server,with_acme,with_clash_api
Revision: f73fb69de8626e935f853e8ebf47e6bc706b1f98
CGO: disabled
```

### Description

Config with dns server of which address is ip without scheme such as `223.5.5.5` won't work, but `udp://223.5.5.5` works. Previous version(`1.9.0-alpha.13`) also works.

### Reproduction

```json
{
    "log": {
        "level": "trace",
        "timestamp": true
    },
    "inbounds": [
        {
            "type": "mixed",
            "listen": "127.0.0.1",
            "listen_port": 12345,
            "sniff": true
        }
    ],
    "outbounds": [
        {
            "type": "direct",
            "tag": "direct"
        },
        {
            "type": "dns",
            "tag": "dns-out"
        }
    ],
    "route": {
        "final": "direct"
    },
    "dns": {
        "servers": [
            {
                "tag": "local-dns",
                "address": "223.5.5.5",
                "detour": "direct"
            }
        ],
        "final": "local-dns"
    }
}
```

### Logs

```shell
FATAL[0000] parse route options: parse dns server[local-dns]: invalid server address
```

### Integrity requirements

- [x] I confirm that I have read the documentation, understand the meaning of all the configuration items I wrote, and did not pile up seemingly useful options or default values.
- [X] I confirm that I have provided the server and client configuration files and process that can be reproduced locally, instead of a complicated client configuration file that has been stripped of sensitive data.
- [X] I confirm that I have provided the simplest configuration that can be used to reproduce the error I reported, instead of depending on remote servers, TUN, graphical interface clients, or other closed-source software.
- [X] I confirm that I have provided the complete configuration files and logs, rather than just providing parts I think are useful out of confidence in my own intelligence.

**Comment by VladimirFedulov** [comment]
The same thing

**Comment by bxyun0** [comment]
The same thing on linux

**Comment by zhuzeitou** [comment]
Fixed in `1.9.0-alpha.15`, close.

**Comment by VladimirFedulov** [comment]
@zhuzeitou profile updating with connected VPN doesn't work. But it works with SB 1.8.6 and 1.9.0-13
![IMG_7461](https://github.com/SagerNet/sing-box/assets/100944929/758c4b14-f162-4a09-a544-34d72664898f)
